src/modules/tests/tests.effects.ts

```diff
@@ -196,9 +197,48 @@ export class TestsEffects {
   );
 
   @Effect()
-  sendTestSuccessEffect$ = this.actions$.pipe(
-    ofType(testActions.SEND_TEST_SUCCESS),
-    switchMap((action: testActions.SendTestSuccess) => {
+  sendCurrentTestEffect$ = this.actions$.pipe(
+    ofType(testActions.SEND_CURRENT_TEST),
+    concatMap(action => of(action).pipe(
+      withLatestFrom(
+        this.store$.pipe(
+          select(getTests),
+        ),
+      ),
+    )),
+    switchMap(([action, tests]: [testActions.SendCurrentTest, TestsModel]) => {
+      const slotId = getCurrentTestSlotId(tests);
+      const test = getCurrentTest(tests);
+
+      if (!test || this.networkStateProvider.getNetworkState() !== ConnectionStatus.ONLINE) {
+        return of(new testActions.SendCurrentTestFailure(slotId));
+      }
+
+      const testToSubmit: TestToSubmit[] = [{
+        slotId,
+        index: 0,
+        payload: test,
+      }];
+
+      return this.testSubmissionProvider.submitTests(testToSubmit)
```

Comment:
Looks like there might be some repetition with the `sendCompletedTestsEffect$` Effect when submitting tests. Could we look to unify them?
Could also have the added benefit of simplifying this flow, currently it has a lot going on with two levels of `switchMap`!